### PR TITLE
Implement MHV verify sign-in widget for supply reorder page

### PR DESCRIPTION
**Pull request by carlosfelix2**
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Did you change site-wide styles, platform utilities or other infrastructure?
- [X] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
This PR creates the new MHV verify sign-in call-to-action widget that will show an alert if the user is unauthenticated or unverified, and otherwise will show any provided content. Initially, this widget is to be used for the [supply reorder page](https://staging.va.gov/health-care/order-medical-supplies/), but it is meant to be reused for other pages too. This new widget uses the same pattern as the CTA widget to be included in a static page.

### Intended Use
For example, the supply reorder page currently has a link to "[Order hearing aid and CPAP supplies online](https://staging.va.gov/health-care/order-hearing-aid-or-CPAP-supplies-form)" that needs to be hidden for users that are not signed in or verified. We will use this widget to hide the link with an alert to make sure users have the proper credentials before accessing the form.

### Widget Features
- Displays an alert instead of content if the user is not signed in
- Displays an alert instead of content if the user is not verified
- Displays optional custom content if no alert is displayed
  -  Custom content is defined under a child element with the `static-widget-content` class (see sample in Template section below). This allows the static content maintainers to update or change the content as needed. 
  - If there is no content then simply nothing is displayed. This allows this widget to be used with or without content. 
  - Note this content comes from an HTML page and therefore must be handled differently in code as it is not a React component
  - For security, custom content HTML is sanitized before being displayed.

### Template for static pages:
The template follows the already stablished [template defined in vets-website](https://github.com/department-of-veterans-affairs/vets-website/blob/cb8e27a144d78ab2b6f7d378468b6d14da4fcb5e/src/platform/landing-pages/dev-template.ejs#L195) and already used in other static pages like the [secure messaging static page](https://staging.va.gov/health-care/secure-messaging/). Here is a specific template example to use this new widget:
```html
<div data-template="paragraphs/react_widget" data-entity-id="13348" data-widget-type="mhv-signin-cta"
  data-widget-timeout="20" data-service-description="order medical supplies">
  <div class="loading-indicator-container">
    <div aria-label="Loading..." aria-valuetext="Loading your application..." class="loading-indicator" role="progressbar">
    </div>
    <span class="loading-indicator-message loading-indicator-message--normal">      
    </span>
    <span class="loading-indicator-message loading-indicator-message--slow vads-u-display--none" aria-hidden="true">
          Sorry, this is taking longer than expected.
    </span>
  </div>
  <span class="static-widget-content vads-u-display--none" aria-hidden="true">
    <p>
      <a class="vads-c-action-link--green" href="https://staging.va.gov/health-care/order-hearing-aid-or-CPAP-supplies-form" hreflang="en">
        Order hearing aid and CPAP supplies online</a>
    </p>
  </span>
  <div class="usa-alert usa-alert-error sip-application-error vads-u-display--none" aria-hidden="true">
    <div class="usa-alert-body">
      <strong>We&#x2019;re sorry. Something went wrong on our end.</strong><br>Please try refreshing your browser.
    </div>
  </div>
</div>
```

- Note the use of `data-widget-type="mhv-signin-cta"` on the first line. All the HTML content under the first div is replaced by the React component which is the widget.
- `data-service-description` is an optional service description string to be added to the headline of the alerts, so they read more specific to the intended call to action
- Also note the content under the span with the class `static-widget-content`. This is the optional content to be displayed if no alert is shown. This content can be zero or more HTML elements.

### How the widget gets rendered in a static page?
Static page widgets are placed in a page by its `create` function. This [vets-website file](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/static-pages/static-pages-entry.js) is responsible for replacing any widgets in static pages based on its widget name as specified by the `data-widget-type` attribute. The template also defines a static loading image that is shown while the React widget is loading, and another static message if there was an issue loading the widget.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/89275

## Testing done
- Added unit tests
- Manual testing of the widget replacement
  - Run a watch on `content-build`. Make sure to use the `cached-assets` to build the repo. This process can take hours.
  - Edit the file `build/localhost/health-care/order-medical-supplies/index.html` in your local `content-build` repo. Replace the 
`Order hearing aid and CPAP supplies online` link with the template shown above in the description section. Save the file. **DO NOT RESTART** the `content-build` watch.
  - Build this branch in `vets-website` with `yarn build --env --entry=static-pages`.
  - Open http://localhost:3002/health-care/order-medical-supplies/ and verify the widget is shown.
- No E2E tests as this widget is not part of any React page.

## Screenshots
**Unauthenticated Alert**
image

**Unverified Alert for non MHV Account**
image

**Unverified Alert for MHV Account**
image

**Show Content when No Alert** - this content is using the sample template above. Optional content is specified as part of the template
image

These AFTER screenshots are only samples as this widget is not yet integrated into the static page(s).

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | localhost_3002_health-care_order-medical-supplies_(iPhone 14 Pro Max)       |   localhost_3002_health-care_order-medical-supplies_(iPhone 14 Pro Max)    |
| Desktop |   localhost_3002_health-care_order-medical-supplies_     |    localhost_3002_health-care_order-medical-supplies_   |

## What areas of the site does it impact?
- Static pages widgets

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback
- Note the addition of the new library called `dompurify`
- Note the use of `dangerouslySetInnerHTML` with `DOMPurify`. There are other examples of using `dangerouslySetInnerHTML` in the repo, but the React community recommends adding `DOMPurify` for security even when we control the static page content.

### CODE REVIEW

File: src/applications/static-pages/mhv-signin-cta/components/messages/CustomAlert.jsx
Comment (carlosfelix2) [inline src/applications/static-pages/mhv-signin-cta/components/messages/CustomAlert.jsx:16]: Note: This custom alert is based on the custom `src/applications/mhv-landing-page/components/MhvRegistrationAlert.jsx`. We need to refactor this file out to a common location and use it for our other alerts once this PR is approved.

File: src/applications/static-pages/mhv-signin-cta/index.js
Comment (carlosfelix2) [inline src/applications/static-pages/mhv-signin-cta/index.js:39]: Note: This content is not a React component and must be handled differently.

File: src/applications/static-pages/mhv-signin-cta/sass/mhv-signin-cta.scss
Comment (carlosfelix2) [inline src/applications/static-pages/mhv-signin-cta/sass/mhv-signin-cta.scss:4]: Note: This SASS is also a copy from `src/applications/mhv-landing-page/sass/mhv-landing-page.scss`. We need to refactor this file out to a common location and use it for our other alerts once this PR is approved.

File: src/applications/static-pages/static-pages-entry.js
Comment (carlosfelix2) [inline src/applications/static-pages/static-pages-entry.js:103]: Note: I will not be fixing any of these cross app imports as it is beyond the scope of this issue.

File: src/applications/static-pages/mhv-signin-cta/index.js
Comment (carlosfelix2) [inline src/applications/static-pages/mhv-signin-cta/index.js:44]: Note: There are other examples of using `dangerouslySetInnerHTML` in the repo, but the React community recommends adding `DOMPurify` for security even when we control the static page content.

File: src/applications/static-pages/mhv-signin-cta/components/messages/CustomAlert.jsx
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/components/messages/CustomAlert.jsx:10]: Typo. Should be `disableAnalytics`

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx:29]: Typically, the naming convention for functions that return jsx/react is `CapitalCase`. Suggesting `defaultAlert` should be named `DefaultAlert`

Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx:29]: Typically, the naming convention for functions that return jsx/react is `CapitalCase`. Suggesting `defaultAlert` should be named `DefaultAlert`

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx:61]: Typically, the naming convention for functions that return jsx/react is `CapitalCase`. Suggesting `mhvAlert` should be named `MhvAlert`

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:30]: We probably want a test for this functionality.

File: src/applications/static-pages/mhv-signin-cta/index.js
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/index.js:49]: I believe these lines could be simplified to:

```
// eslint-disable-next-line react/no-danger
return (<div dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(noAlertContent)}} />);
```

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx:29]: Fixed. Thx.

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx:61]: Fixed. Thx.

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:30]: I'll look into this, but I have been unsure how I test something like this that imports and renders. There are no existing examples.

Comment (radavis) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:30]: Then, I don't understand why this component prop exists.

File: src/applications/static-pages/mhv-signin-cta/index.js
Comment (radavis) [reply src/applications/static-pages/mhv-signin-cta/index.js:66]: When visiting va.gov as an unauthenticated user, these slices of state should return falsy values.

Screenshot from 2024-08-13 11-03-50

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:33]: Added new unit test

File: src/applications/static-pages/mhv-signin-cta/index.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/index.js]: Ok. Simplified the code.

File: src/applications/static-pages/mhv-signin-cta/test/MhvSigninCallToAction.unit.spec.jsx
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/test/MhvSigninCallToAction.unit.spec.jsx:131]: Suggestion: Since your unit specs for these components already have assertions for headings, links, and analytics you could make a more-general, static assertion about the component being rendered, such as with testIds. No change necessary.

**Review by radavis — COMMENTED** [review]
I still need to boot this up, locally. But, I wanted to leave some feedback while I'm waiting for the content-build process to complete.

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx:30]: Suggestion: It would be less confusing if the HTML provided the full headline, or the component provided its own.

File: src/applications/static-pages/mhv-signin-cta/test/createMhvSigninCallToAction.unit.spec.jsx
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/test/createMhvSigninCallToAction.unit.spec.jsx:68]: This test asserts that `ReactDOM.render` was called with arguments. If you want to verify the outcome/output, use the [`document.getElement*` methods](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByClassName).

I think this test could be simplified to just checking that `ReactDOM.render` was called and the document has the element with id/class/data-* that you expect.

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnauthenticatedAlert.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/components/messages/UnauthenticatedAlert.jsx:22]: I think it is worth checking. I got this code from another app, so I am wondering how that works.

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:8]: The problem I have is that BABEL_ENV is not something set by the Mocha test framework. We would be using a variable meant for some other tool. Babel is a JS compiler.

File: src/applications/static-pages/mhv-signin-cta/test/createMhvSigninCallToAction.unit.spec.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/test/createMhvSigninCallToAction.unit.spec.jsx:68]: The document does not change after that call. I was hoping that would work, but had to revert to just checking the call is correct. I think this has to do with the way it is rendered in the static page and the React wrapper they add, which is not added as part of this code.

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx:30]: Not sure what you mean. We have different alerts that have different headlines, but the service description is shared.

File: src/applications/static-pages/mhv-signin-cta/test/MhvSigninCallToAction.unit.spec.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/test/MhvSigninCallToAction.unit.spec.jsx:131]: I see. I'll take a look.

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:33]: I hear you, but they have added already other custom attributes. I was trying not to introduce yet another way to pass more information. I'll take a look and see how this looks.

Comment (radavis) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:33]: What I'm trying to convey is that instead of:

html: `<div service-description="a value" />`
js: `el.getAttribute('service-description')`

do the following

html: `<div data-service-description="a value" />`
js: `el.dataset.serviceDescription` or `el.getAttribute('data-service-description')`

if possible.

File: src/applications/static-pages/mhv-signin-cta/test/MhvSigninCallToAction.unit.spec.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/test/MhvSigninCallToAction.unit.spec.jsx:131]: This makes sense and I changed the code. I did test for the service description text to make sure it is passed down to the components.

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/components/messages/UnverifiedAlert.jsx:30]: I looked into this more. The problem is that the headline is set by the template provided by CAIA and the only thing that can change in the headline is the service description. Here are the templates: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content/content-patterns-and-standards/sign-in-alerts.md#verify-alert-for-loa1ial1-logingov-and-idme-accounts.

File: src/applications/static-pages/mhv-signin-cta/components/messages/UnauthenticatedAlert.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/components/messages/UnauthenticatedAlert.jsx]: I looked into this some more. The trigger is a string to identify who triggered the modal. This seems to be me as an analytics string, and also barely any apps use it. Those that use it just add some string that is unrelated to anything like app name, etc. The derivedUrl is where to redirect, which the default is what we want. Thx for bringing this up.

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (radavis) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:8]: Here are a few ways to determine if Mocha is running: https://stackoverflow.com/questions/29183044/how-to-detect-if-a-mocha-test-is-running-in-node-js

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:29]: Is this an alias for array indexing, such as `[0]`?

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (radavis) [inline src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:33]: I would decouple getting this data from calling the component so we get a clear error message if something goes wrong:

```
const serviceDescription = el.getAttribute('data-service-description');

...

<MhvSigninCallToAction serviceDescription={serviceDescription}
...
```

**Review by radavis — DISMISSED** [review]
Approved w/ comments

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:29]: Yes and no. item(0) will return null if there is nothing there. [0] will throw an exception. Using item means I don't have to make other checks to safeguard that.

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:33]: Makes sense.

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:29]: Although now testing it it does not care either way. I must have had some other code that broke before I added that

File: src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js:8]: Hmmm... Still not convinced. The correct way is to tell mocha to not to use CSS/SASS. I found the config for this, but need to pass that by Platform before making changes to such a global config.

Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/createMhvSigninCTA.js]: Oh! This just occurred to me: I just need to move the import a level higher and then we will not need this check. Still, I want to pursue getting that config to ignore sass.

File: src/applications/static-pages/static-pages-entry.js
Comment (carlosfelix2) [inline src/applications/static-pages/static-pages-entry.js:104]: Importing this SASS here allow us to unit test the other files as @imports breaks the unit test framework.

**Review by radavis — DISMISSED** [review]

**Review by sterkenburgsara — COMMENTED** [review]
@carlosfelix2 initial sign-in widget (not verification one) should be the blue info-alert styling instead of green.

**Review by sterkenburgsara — COMMENTED** [review]
Looks great. Thank you for making this fix!

**Review by radavis — APPROVED** [review]

File: src/applications/static-pages/mhv-signin-cta/components/messages/CustomAlert.jsx
Comment (carlosfelix2) [reply src/applications/static-pages/mhv-signin-cta/components/messages/CustomAlert.jsx:45]: Such a good boy! 🐶

**Comment by carlosfelix2** [comment]
Waiting on approval to add dompurify

**Review by powellkerry — APPROVED** [review]

**Review by CBonade — APPROVED** [review]

**Comment by carlosfelix2** [comment]
From @CBonade: You’re good to move ahead - POAM ticket is being written security will take care of that

**Comment by laurwill** [comment]
Hi @sterkenburgsara @carlosfelix2 , we made some significant content changes to these widgets over the past month as part of the work to [standardize this pattern in VADS](https://www.figma.com/design/afurtw4iqQe6y4gXfNfkkk/VADS-Component-Library?node-id=13356-1783&node-type=frame&t=0WOZbLpX2YqGxv0G-0). It looks like you may be finished with implementation here — but maybe you could open a backlog ticket to update to the current content? Apologies for the ever-changing content in these, and let me know if you want to chat!

**Review by radavis — COMMENTED** [review]
Please disregard this message. :)

**Pull request merged**
